applications/order/groovyScripts/entry/BillSettings.groovy

```diff
@@ -66,20 +66,20 @@ if (cart) {
         checkOutPaymentId = cart.getPaymentMethodIds().get(0)
         context.checkOutPaymentId = checkOutPaymentId
         if (!orderParty) {
-            paymentMethod = from("PaymentMethod").where("paymentMethodId", checkOutPaymentId).queryOne()
-            if ("CREDIT_CARD".equals(paymentMethod?.paymentMethodTypeId)) {
-                paymentMethodType = "CC"
-                account = paymentMethod.getRelatedOne("CreditCard", false)
+            paymentMethod = from('PaymentMethod').where('paymentMethodId', checkOutPaymentId).queryOne()
+            if (paymentMethod?.paymentMethodTypeId == 'CREDIT_CARD') {
+                paymentMethodType = 'CC'
+                account = paymentMethod.getRelatedOne('CreditCard', false)
                 context.creditCard = account
                 context.paymentMethodType = paymentMethodType
-            } else if ("EFT_ACCOUNT".equals(paymentMethod.paymentMethodTypeId)) {
-                paymentMethodType = "EFT"
-                account = paymentMethod.getRelatedOne("EftAccount", false)
+            } else if (paymentMethod.paymentMethodTypeId == 'EFT_ACCOUNT') {
```

Comment:
Not related to codenarc application but there is a potential NPE to check
(paymentMethod. instead of paymentMethod?.)

Reply:
Ok with the improvment